gui/src/app/pages/HomePage/SaveProjectWindow.tsx

```diff
@@ -0,0 +1,213 @@
+import { FunctionComponent, useCallback, useContext, useState } from "react";
```

Comment:
Just noting that this file is getting a little chunky--might consider splitting into separate files for the major components at some point.

Reply:
agreed